ledger/daml-on-sql/src/main/scala/on/sql/Main.scala

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.on.sql
+
+import com.daml.ledger.participant.state.v1
+import com.daml.ledger.participant.state.v1.SeedService.Seeding
+import com.daml.platform.configuration.{InvalidConfigException, LedgerConfiguration}
+import com.daml.platform.sandbox.cli.Cli
+import com.daml.platform.sandbox.config.{LedgerName, SandboxConfig}
+import com.daml.platform.sandbox.{GlobalLogLevel, SandboxServer}
+import com.daml.resources.ProgramResource
+
+object Main {
+
+  private val Name = LedgerName("DAML-on-SQL")
+
+  private[sql] val defaultConfig: SandboxConfig =
```

Comment:
Ideally we could make the `--ledgerid` parameter required for daml-on-sql. Maybe pass that in as a parameter to `Cli`?

Reply:
We can check if it's set in `Main` right below here. Perhaps that'd be easier to understand.

TBH I am not a fan of making `--switch`-style switches mandatory, but we can't really do otherwise without changing the API.

Reply:
Done.

Reply:
What do you think about removing all post-config checks from `Main` and rather parametrizing `Cli` accordingly, e.g. make `--sql-backend-jdbcurl` required, remove (or hide) `--implicit-party-allocation`, etc... ?

Reply:
TBH I think it would hide the logic custom to the DAML-on-SQL product. Putting a bunch of `if` checks in `Cli` seems to me to be the wrong place for the responsibility.

I pulled out a separate Bazel package and main class for DAML-on-SQL so that I could avoid a bunch of `if` checks scattered around the code. By adding the package and putting all custom behavior in the main function, there's only one `if`, and it's in the user's brain when they choose which ledger implementation to run.

Reply:
Hmmm, thinking about this again, you're right that the `Cli` is better-placed to yield more useful error messages for less code. I'll try it and see how it goes.

Reply:
Opened https://github.com/digital-asset/daml/pull/6846 as an attempt. I don't really like it but I can't see a much better approach IMO.

Most of the additions are extra tests.

Reply:
Merged #6846 into this one. Can I get a review please?